include/faabric/util/dirty.h

```diff
@@ -25,24 +25,22 @@ class DirtyTracker
   public:
     virtual void clearAll() = 0;
 
-    virtual void reinitialise() = 0;
+    virtual std::string getType() = 0;
```

Comment:
Could we have add some lines to the class description comment (in lines 20-21) to add the different types of tracking methods and a very brief explanation?

I know this is also explained in each subclass description, but I think it would be helpful to have all information summarised together somewhere.